### PR TITLE
Feat: Footer component

### PR DESCRIPTION
**Pull request by alcferreira**
ISSUES CLOSED: #79

### CODE REVIEW

File: packages/footer/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  "bugs": {
+    "url": "https://github.com/CraveFood/farmblocks/issues"
+  },
+  "homepage": "https://github.com/CraveFood/farmblocks#readme",
+  "keywords": ["farmblocks", "footer", "react", "component"],
+  "dependencies": {
+    "@crave/farmblocks-button": "^1.0.7",
+    "@crave/farmblocks-text": "^1.1.0",
+    "@crave/farmblocks-theme": "^1.4.0"
```
[the remark this answers is not in the document]
Comment (alcferreira) [reply packages/footer/package.json]: Oops!

File: packages/footer/src/styledComponents/Footer.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import styled from "styled-components";
+import { colors as colorConstants } from "@crave/farmblocks-theme";
+
+const margin = "16px";
+
+const FooterContainer = styled.div`
+  font-family: Lato, sans-serif;
```
Comment (fczuardi) [inline packages/footer/src/styledComponents/Footer.js]: we need to remember to remove this when/if we replace the a link with a future Link component

File: packages/footer/src/components/Footer.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        )}
+      </div>
+    </Container>
+  );
+};
+
+Footer.propTypes = {
+  helpText: PropTypes.string.isRequired,
+  helpLinkText: PropTypes.string.isRequired,
+  helpLinkHref: PropTypes.string.isRequired,
```
Comment (fczuardi) [inline packages/footer/src/components/Footer.js]: same as below:
> should linkText and linkHref be required? what if the helper text alone is enough to explain the content that needs help?

File: packages/footer/src/styledComponents/Footer.js (hunk 2 above)
Comment (alcferreira) [reply packages/footer/src/styledComponents/Footer.js]: Created https://github.com/CraveFood/farmblocks/issues/121 to address this

**Comment by alcferreira** [comment]
@fczuardi all done to review again please.

File: packages/footer/src/components/Footer.js [hunk 4]
```diff
@@ -48,8 +48,8 @@ const Footer = props => {
 
 Footer.propTypes = {
   helpText: PropTypes.string.isRequired,
-  helpLinkText: PropTypes.string.isRequired,
-  helpLinkHref: PropTypes.string.isRequired,
+  helpLinkText: PropTypes.string,
+  helpLinkHref: PropTypes.string,
```
Comment (fczuardi) [inline packages/footer/src/components/Footer.js]: 👍

File: packages/footer/src/components/HelperContent.js [hunk 5]
```diff
@@ -6,31 +6,34 @@ import Image, { badgeSizes } from "@crave/farmblocks-image";
...
+    </Container>
+  );
+};
 
 HelperContent.propTypes = {
   text: PropTypes.string.isRequired,
-  linkText: PropTypes.string.isRequired,
-  linkHref: PropTypes.string.isRequired
+  linkText: PropTypes.string,
+  linkHref: PropTypes.string
```
Comment (fczuardi) [inline packages/footer/src/components/HelperContent.js]: 👍

File: packages/footer/src/components/HelperContent.js [hunk 6]
```diff
@@ -31,9 +27,15 @@ const HelperContent = props => {
...
 HelperContent.propTypes = {
+  imageSrc: PropTypes.string,
   text: PropTypes.string.isRequired,
   linkText: PropTypes.string,
   linkHref: PropTypes.string
 };
 
+HelperContent.defaultProps = {
+  imageSrc:
+    "https://crave-whatsgood-sandbox.imgix.net/businesses/32/inventory/8fae5d32-f6d4-47bb-8062-e4e85c47788b.png"
```
Comment (fczuardi) [inline packages/footer/src/components/HelperContent.js]: are we ok with the possibility of other projects hotlinking our assets, and spending our bandwidth? not a big deal at the moment, but maybe we could study the possibility of embeding the contents of the default image instead.

for example:

```
imageSrc: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAHgAAAB4CAYAAAA5ZDbSAAALQ0lEQVR4nO2dfWwT5x3Hvz7H9iWx40tCEieEJJSuFBIIoWiFQYlakLKpbKNMpWisTIAm2n9aunUV2sRfSAit1RjdpLZ/EDSqagU2FqllAq2lpU0HtJCXhjUd4yUJie0sb3634/PL/kjT0txzxuecfZdHz+fPs333PP7cPa+/5znDqtf/mwSDWjitE8DILkww5TDBlMMEUw4TTDlMMOUwwZTDBFMOE0w5TDDlMMGUwwRTDhNMOUww5TDBlMMEUw4TTDlMMOUwwZTDBFMOE0w5TDDlMMGUwwRTDhNMOUww5TDBlMMEUw4TTDl5WidAbSxGAxodPJaUWVArmFEnmFFaYESBiUOhaep+DooJhMQERoMx9HtF9Hui6B2ZRLc7gsk4XYstqRAs8Bxa7rfhsYVWLKvgYTIaUn/faITAG1FlM2G5I//r42I8iZ7hCM7fDuDcDT88kUS2k551DHN5fXCjg8fTjQLW1hQij0stVSmxRBKfDATxZrcH3e6IqufOJXNScFMljz2rSvFQVf69v6wCV51hvP7ZGLrmoOg5Jbgk34jnVpfi8QeKNLn+mes+vHppDOPhuCbXz4Q5I/jh6nwc3OhAkcWY9m+GAzH0eaIY8onwTU41rACgwMShyMJhfpEJdYIZFdb0myK+yTh+854blwfDivOgBXNC8FMNdrywZh6M96hn44kkutwRXOgL4EJfEE5/LK3zV9ny0FxXiOY6K1Y4+LSuc/jiKE5c86adB63QtWAjB+xbV4bNS+wpv+fyi2jtmMAHtwPwTs6u5Wu3cHh0oRW7Vhaj0mZK+d22Xi8OtY8gruPGtm4F2y0cXm6pRFOlfEPKG4njWOcETl7zQFT5TzZxwNYGATubimHn5auFTlcYvz7nmvWNlS10KdjIAa9tmi8rN5lM4q3PPTjaMYFANLt/rNXMYffKYmxfLsBgIBfdna4wnn13SJdPsi6HKvetK5OVGxITePGcG0cujWVdLgAEogkcuTSGF8+5v26kzaSpMh/71pVlPS2ZoDvB2xrssnWu0y9iV9sgPuoP5jhVwEf9QexqG4TTLxI/37zEjm0NqdsKWqArwQ9X52PvmnnEz3qGI9hx+g5ujkdznKpvuDkexY7Td9AzTB7w2LtmHh6uzs3gS7roRnBJvhEHNzqIXRSnX8QLZ53w6mBs2BtJ4IWzTuKTbOQMOLjRgZL89Pvq2UY3gp9bXUocxAiJCfzyrEsXcqfxRqbSRKqTiyxTo216QRezSU2VPHH4MZlMYv/7wxkVywLPYXV1Ab5bXYDvlFpQzBtRnG9ELJGEbzIOlz+GnuEILt4J4YpT+ajUzfEo9r8/jFdaHJLW9eMPFKGt16eLsWtddJPe+OF8rCRMHLzZPYFXL40pOleVLQ8/X1GMTYuLYL7HtOE0/Z4o/nR5DB/2KW+8Pb+6FD9rLJYcv+oM45l3hhSfT200L6IbHTxRrjcSR2vHhKJzPVlvx4mtNdiy1J62XACoFcx4uaUS+5vLoXTS8WjHBLwR6eTDQ1X5aHTwCs+mPpoLfrpRIB5v7VQ2iLFvXRleWlcGPi/zLP3owSL8trlc0W8C0QSOdZJvRLm85RJNBQs8h7U1hZLjLr+IU9c8aZ+n0cHjJ/Xq9EF//GARvregQNFvTl7zwEVoVa+tKYTAa/sMaXr1lvttxEiM1o4JRWPLS8ssKqYK2LOqRNH3xQSI1UkeZ0DL/Ta1kpURmraiH1tolRyLJ5L44HZA0Xn6POTRJQAY8ol472YAtyaiGPBGUWjm8ECpBU81CLLzwEvLeTiseXAH0ptuBIAPbgew75EyST/+sYVWTacVNRNsyTNgWYW0EdLlDiuembl4J4SP+4N4pPab4t4TjuNoxzj++oUXsRmnuzwYxt97ffjzlmrU2M3EczaU83AH0r/RvJMJdLkjkjCiZRU8LEaDZtGamglulIl+vJBBVwUAfnXWhU2Lbagv43HHJ6LtSx+CKRppgWgCb/d48ZLMJEFpgfLRqAt9AYlg01dhvJ8OaRMBolkdvLSM3IXIVHASwDv/8eNQ+wje+tyTUu40wymKYCXdrGnk0r5E5TaCEjQTXCtIoyXcATHtMBs1qC+X/+MnMgisc/pjxJumViBXA7lAQ8HSTPenaCypTVMlj58uF2Q/vzWR2axVn0f6uzoNBWtWB5PquEFvdgXn5xlQX87jiSVF2HCfVTa4bjQYwxcjkxldY8gnzUMm9blaaCa4wCQtPPxZiNAoNHPYs6oEa6oLUCOYwMmE3dzNsS5lQ6R34yP0AEh5zRWaCS4kZFouJGY27G8ux4b7pP1tOT4dDOFv/86830rKAymvuUIX04XZREkL9p83/Tjw4f9A0wJDzQQHxQQE47frpmwUZRPhOKruEd884I3id+0jqqxWIOUhmIWSKV00ExwSExBmxBsXWdQXPOAVUV8u7XM7/SLa+4P4uD+EK86QZLQrU0h5yEbVky6aCR4LSZ+s+UWpn7RMGCB0W64MhfDsu07VrwUA1YQ8jAZz17efiWa1f3+O+ouXCUOE7QMh1a8zDbF/n+XuXyo0E0waEKiw5qHKpm6h0jMcwR8vjyIQTSAaT+LMdR/eVjDXrIQqWx5xhop0M+cKzYroXpmBhOa6QvylR93pteNdHhzv8oAzAIkstpCb66TBC4B8XnOBZk9wtzsCkdAfaa5Lv8+qlGzKBchpF+NJTbeA0Ezw5FcbnsxkhYOHPQutaXV38JBit3BYQQiy6xnWduceTQc6zt8OSCIqjZwBjy60ou1LnyrXWFxqxt4189BQziOWBDqcYfz+XyMYUnnW6tGF5LHt8wqjU9RG05isczf8iBHKzV0ri6HGmEetYMLRzdVYNb8AvImD1cxhfV0hjj2xAMUp1vwqxcRNpXkmsUQS5274VbtOJmgq2BNJ4JMB6SR5pc2ErQ3CrM//i4dKYCGE0RbnG1NOFSpla4NA3A3gk4Gg5nttaR4X/Wa3h3h8Z1MxrObZJW/xPPlx6AdTfKYEq5nDzibp0wvI5y2XaC642x3BVcLaIDtvxG5CsaeE8ZB8VMZYWJ06ePdK8hYPV51hXWygprlgAHjjCnn90fblAtbXkvuW6ZCqgXP+1uwXka+vLcR2maL+9c+UranKFroQ3OmK4Mx1aavZYDDgwIYKLCrJbAjz5DWv5LyJZBLHOsdnvUvAohIzDmyoIO7bcea6PlYWAjpZXQhMLQA/9VQNcY2w0y9ix+k7Ga8Rri+3YHkFj1gC6HCFZ71LgJ3ncHzLAuI0pG8yjidPDOhmNzzdCAamtnA48oMqYn+yZziii1X+dp7D4e9XEYP244kknvuHU7MYaBK6KKKnuTwYxuGLo8TPllXwOL5lQcbFtRosKjHj+JYFRLkAcPjiqK7kAjoTDAAnrnnR1kuebKiymdC6uXpWDa9MWV9biNbN1bLRIW29Xl1ubag7wQBwqH0EnS7yk1Bg4vBKiwPPry6ddT85HaxmDs+vLsUrLQ7ZkKJOVxiH2keynpZM0FUdfDfpbmXY2jmBU1nayvDJBgG72FaG2YNtRjp7dC14mm0NduzV2XbCf7g4ird1WOfOZE4IBma3IfigV4Q/+u0NwW1mDtV2tiG4rmBb+itnTgmepqmSxzOrSonbL2WDDmcYr7GXcuQe9lqdezOn1yZ1uyPodrsVvxhLDvZirDnA3a+2qxPMqE3xaruxUBz9nij62Kvt5g6T8SQ+HQrrbkxYK3Q5VMlQDyaYcphgymGCKYcJphwmmHKYYMphgimHCaYcJphymGDKYYIphwmmHCaYcphgymGCKYcJphwmmHKYYMphgimHCaYcJphymGDKYYIphwmmHCaYcphgymGCKYcJppz/A1DjWI1TIQvqAAAAAElFTkSuQmCC"
```

File: packages/footer/src/components/HelperContent.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+HelperContent.propTypes = {
+  imageSrc: PropTypes.string,
+  text: PropTypes.string.isRequired,
+  linkText: PropTypes.string,
+  linkHref: PropTypes.string
+};
+
+HelperContent.defaultProps = {
+  imageSrc:
+    "https://crave-whatsgood-sandbox.imgix.net/businesses/32/inventory/8fae5d32-f6d4-47bb-8062-e4e85c47788b.png"
```
Comment (fczuardi) [inline packages/footer/src/components/HelperContent.js]: are we ok with the possibility of other projects hotlinking our assets, and spending our bandwidth? not a big deal at the moment, but maybe we could study the possibility of embeding the contents of the default image instead.

for example:

```
imageSrc: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAHgAAAB4CAYAAAA5ZDbSAAALQ0lEQVR4nO2dfWwT5x3Hvz7H9iWx40tCEieEJJSuFBIIoWiFQYlakLKpbKNMpWisTIAm2n9aunUV2sRfSAit1RjdpLZ/EDSqagU2FqllAq2lpU0HtJCXhjUd4yUJie0sb3634/PL/kjT0txzxuecfZdHz+fPs333PP7cPa+/5znDqtf/mwSDWjitE8DILkww5TDBlMMEUw4TTDlMMOUwwZTDBFMOE0w5TDDlMMGUwwRTDhNMOUww5TDBlMMEUw4TTDlMMOUwwZTDBFMOE0w5TDDlMMGUwwRTDhNMOUww5TDBlMMEUw4TTDl5WidAbSxGAxodPJaUWVArmFEnmFFaYESBiUOhaep+DooJhMQERoMx9HtF9Hui6B2ZRLc7gsk4XYstqRAs8Bxa7rfhsYVWLKvgYTIaUn/faITAG1FlM2G5I//r42I8iZ7hCM7fDuDcDT88kUS2k551DHN5fXCjg8fTjQLW1hQij0stVSmxRBKfDATxZrcH3e6IqufOJXNScFMljz2rSvFQVf69v6wCV51hvP7ZGLrmoOg5Jbgk34jnVpfi8QeKNLn+mes+vHppDOPhuCbXz4Q5I/jh6nwc3OhAkcWY9m+GAzH0eaIY8onwTU41rACgwMShyMJhfpEJdYIZFdb0myK+yTh+854blwfDivOgBXNC8FMNdrywZh6M96hn44kkutwRXOgL4EJfEE5/LK3zV9ny0FxXiOY6K1Y4+LSuc/jiKE5c86adB63QtWAjB+xbV4bNS+wpv+fyi2jtmMAHtwPwTs6u5Wu3cHh0oRW7Vhaj0mZK+d22Xi8OtY8gruPGtm4F2y0cXm6pRFOlfEPKG4njWOcETl7zQFT5TzZxwNYGATubimHn5auFTlcYvz7nmvWNlS10KdjIAa9tmi8rN5lM4q3PPTjaMYFANLt/rNXMYffKYmxfLsBgIBfdna4wnn13SJdPsi6HKvetK5OVGxITePGcG0cujWVdLgAEogkcuTSGF8+5v26kzaSpMh/71pVlPS2ZoDvB2xrssnWu0y9iV9sgPuoP5jhVwEf9QexqG4TTLxI/37zEjm0NqdsKWqArwQ9X52PvmnnEz3qGI9hx+g5ujkdznKpvuDkexY7Td9AzTB7w2LtmHh6uzs3gS7roRnBJvhEHNzqIXRSnX8QLZ53w6mBs2BtJ4IWzTuKTbOQMOLjRgZL89Pvq2UY3gp9bXUocxAiJCfzyrEsXcqfxRqbSRKqTiyxTo216QRezSU2VPHH4MZlMYv/7wxkVywLPYXV1Ab5bXYDvlFpQzBtRnG9ELJGEbzIOlz+GnuEILt4J4YpT+ajUzfEo9r8/jFdaHJLW9eMPFKGt16eLsWtddJPe+OF8rCRMHLzZPYFXL40pOleVLQ8/X1GMTYuLYL7HtOE0/Z4o/nR5DB/2KW+8Pb+6FD9rLJYcv+oM45l3hhSfT200L6IbHTxRrjcSR2vHhKJzPVlvx4mtNdiy1J62XACoFcx4uaUS+5vLoXTS8WjHBLwR6eTDQ1X5aHTwCs+mPpoLfrpRIB5v7VQ2iLFvXRleWlcGPi/zLP3owSL8trlc0W8C0QSOdZJvRLm85RJNBQs8h7U1hZLjLr+IU9c8aZ+n0cHjJ/Xq9EF//GARvregQNFvTl7zwEVoVa+tKYTAa/sMaXr1lvttxEiM1o4JRWPLS8ssKqYK2LOqRNH3xQSI1UkeZ0DL/Ta1kpURmraiH1tolRyLJ5L44HZA0Xn6POTRJQAY8ol472YAtyaiGPBGUWjm8ECpBU81CLLzwEvLeTiseXAH0ptuBIAPbgew75EyST/+sYVWTacVNRNsyTNgWYW0EdLlDiuembl4J4SP+4N4pPab4t4TjuNoxzj++oUXsRmnuzwYxt97ffjzlmrU2M3EczaU83AH0r/RvJMJdLkjkjCiZRU8LEaDZtGamglulIl+vJBBVwUAfnXWhU2Lbagv43HHJ6LtSx+CKRppgWgCb/d48ZLMJEFpgfLRqAt9AYlg01dhvJ8OaRMBolkdvLSM3IXIVHASwDv/8eNQ+wje+tyTUu40wymKYCXdrGnk0r5E5TaCEjQTXCtIoyXcATHtMBs1qC+X/+MnMgisc/pjxJumViBXA7lAQ8HSTPenaCypTVMlj58uF2Q/vzWR2axVn0f6uzoNBWtWB5PquEFvdgXn5xlQX87jiSVF2HCfVTa4bjQYwxcjkxldY8gnzUMm9blaaCa4wCQtPPxZiNAoNHPYs6oEa6oLUCOYwMmE3dzNsS5lQ6R34yP0AEh5zRWaCS4kZFouJGY27G8ux4b7pP1tOT4dDOFv/86830rKAymvuUIX04XZREkL9p83/Tjw4f9A0wJDzQQHxQQE47frpmwUZRPhOKruEd884I3id+0jqqxWIOUhmIWSKV00ExwSExBmxBsXWdQXPOAVUV8u7XM7/SLa+4P4uD+EK86QZLQrU0h5yEbVky6aCR4LSZ+s+UWpn7RMGCB0W64MhfDsu07VrwUA1YQ8jAZz17efiWa1f3+O+ouXCUOE7QMh1a8zDbF/n+XuXyo0E0waEKiw5qHKpm6h0jMcwR8vjyIQTSAaT+LMdR/eVjDXrIQqWx5xhop0M+cKzYroXpmBhOa6QvylR93pteNdHhzv8oAzAIkstpCb66TBC4B8XnOBZk9wtzsCkdAfaa5Lv8+qlGzKBchpF+NJTbeA0Ezw5FcbnsxkhYOHPQutaXV38JBit3BYQQiy6xnWduceTQc6zt8OSCIqjZwBjy60ou1LnyrXWFxqxt4189BQziOWBDqcYfz+XyMYUnnW6tGF5LHt8wqjU9RG05isczf8iBHKzV0ri6HGmEetYMLRzdVYNb8AvImD1cxhfV0hjj2xAMUp1vwqxcRNpXkmsUQS5274VbtOJmgq2BNJ4JMB6SR5pc2ErQ3CrM//i4dKYCGE0RbnG1NOFSpla4NA3A3gk4Gg5nttaR4X/Wa3h3h8Z1MxrObZJW/xPPlx6AdTfKYEq5nDzibp0wvI5y2XaC642x3BVcLaIDtvxG5CsaeE8ZB8VMZYWJ06ePdK8hYPV51hXWygprlgAHjjCnn90fblAtbXkvuW6ZCqgXP+1uwXka+vLcR2maL+9c+UranKFroQ3OmK4Mx1aavZYDDgwIYKLCrJbAjz5DWv5LyJZBLHOsdnvUvAohIzDmyoIO7bcea6PlYWAjpZXQhMLQA/9VQNcY2w0y9ix+k7Ga8Rri+3YHkFj1gC6HCFZ71LgJ3ncHzLAuI0pG8yjidPDOhmNzzdCAamtnA48oMqYn+yZziii1X+dp7D4e9XEYP244kknvuHU7MYaBK6KKKnuTwYxuGLo8TPllXwOL5lQcbFtRosKjHj+JYFRLkAcPjiqK7kAjoTDAAnrnnR1kuebKiymdC6uXpWDa9MWV9biNbN1bLRIW29Xl1ubag7wQBwqH0EnS7yk1Bg4vBKiwPPry6ddT85HaxmDs+vLsUrLQ7ZkKJOVxiH2keynpZM0FUdfDfpbmXY2jmBU1nayvDJBgG72FaG2YNtRjp7dC14mm0NduzV2XbCf7g4ird1WOfOZE4IBma3IfigV4Q/+u0NwW1mDtV2tiG4rmBb+itnTgmepqmSxzOrSonbL2WDDmcYr7GXcuQe9lqdezOn1yZ1uyPodrsVvxhLDvZirDnA3a+2qxPMqE3xaruxUBz9nij62Kvt5g6T8SQ+HQrrbkxYK3Q5VMlQDyaYcphgymGCKYcJphwmmHKYYMphgimHCaYcJphymGDKYYIphwmmHCaYcphgymGCKYcJphwmmHKYYMphgimHCaYcJphymGDKYYIphwmmHCaYcphgymGCKYcJppz/A1DjWI1TIQvqAAAAAElFTkSuQmCC"
```

**Pull request merged**